### ISSUE TITLE
Bad multi-monitor support and issues

### ISSUE THREAD

**Issue by Keziolio** [teeworlds/teeworlds#1402]
Hi, i have 3 monitors on my system, when starting the game, the game stretches to fill all the 3 monitors (that's right)
But:

1) the UI is stretched, you should put a limit on how wide the UI can be, or put a slider/option, like in Serious Sam 3, ideally the UI should fit on the central monitor

![](http://images.akamai.steamusercontent.com/ugc/431573485396435498/76378CCFF024B28BA6D65057B341ECA221186DE0/)

2) the game is stretched, the field of view is vertically too narrow, the game is unplayable on 3 monitors, here's how it looks:

![](http://images.akamai.steamusercontent.com/ugc/431573485396418091/BA06354F8FA003DD37DBB436DABAE41688306561/?interpolation=lanczos-none&output-format=jpeg&output-quality=95&fit=inside|2048:384&composite-to=*,*|2048:384&background-color=black)
(The charachter also looks pixellated)

You need to keep the vertical field of view constant, not the horizontal one, you are making the life impossible to everyone that has a setup wider than a 16:9 (including 2-3 monitors setup and 21:9 monitors, that are getting more and more common)

3) when the resolution is set to 1920x1080, the game shows on the left monitor, and the center/right monitor are black, ideally you want to be fullscreen on ONE monitor, (maybe the center/main one), and the others should be at least usable. Or at least put the game on the center one, you don't want to get stiff neck playing with the left monitor.
The only fix for this is playing with the game windowed, this is just annoying.

4) On Arch Linux, i can't alt-tab or make anything outside the game, this is too 1990ish, please fix

**Comment by heinrich5991** [comment]
Work-around for at least 4): Enable borderless window mode.

**Comment by TsFreddie** [comment]
Teeworlds has extreme limit on field of view since larger horizontal view would be unfair to other 16:9 or 4:3 players. BTW, 5:4 ratio (that supported by default) has the best field of view.

**Comment by Keziolio** [comment]
@heinrich5991 
Thanks, borderless window should fix something


@TsFreddie 
This is the same as if i play with a 5:4, i have an unfair advantage over 16:9 players, if you put it in this way.
The market is going towards wider and wider screens, not the opposite, you are putting a limit on what is already became a standard, and is going to grow (people are happy with 21:9 monitors)

Nobody is using 5:4 ratio monitors since 10 years, this is a pointless limitation. And it's making the game unusable.

**Comment by TsFreddie** [comment]
@Keziolio Yes, it does have an unfair advantage over 16:9, but most maps (especially team map) have a horizontal layout. I'm not really sure, but I think the absolute largest field of view would appeared at 1:1 ratio (due to horizontal limitation). It's just a way to limit players view along with the map design.

**Comment by TsFreddie** [comment]
We may place a ratio option to let player choose their ratio and make other part of the rendering surface black in order to support super wide resolution and maybe unblock the black part when spectating.

**Comment by dimidrolqq** [comment]
I have a different minor issue with multiple monitors, not sure if it's worth a new ticket, so i'll just post it here.
I run the game only on one of the monitors in borderless mode. While doing the same in other games i get the behaviour that when the game is "paused" or in some kind of menu or whatever, the mouse can be moved out of the game window. In tw this isn't possible, you have to alt+tab to be able to get out of the window

**Comment by MegaManSec** [comment]
There used to be a setting that would allow for what you're talking about. It was something like inp_grab, or input_grab, or something like that. I found that in windowed mode, moving the mouse very fast would allow the mouse to 'leak out' of Teeworlds, which of course caused problems. setting the inp_grab variable to '1', would fix this, and it would also mean I couldn't move the mouse outside of the window mode when in a menu or whatever. So Perhaps try setting inp_grab to 0, and trying.

Just a thought.

**Comment by heinrich5991** [comment]
@dimidrolqq Another work-around for you: Pressing f1 for the local console will allow you to move the mouse out of the window.

**Comment by dimidrolqq** [comment]
@MegaManSec Good thought, it is unfortunately already 0 by default though.
@heinrich5991 That works
Thanks to both of you

**Comment by oy** [comment]
Fixed.
